src/index.ml

```diff
@@ -753,7 +755,6 @@ struct
     | None -> ()
     | Some log ->
         Tbl.iter f log.mem;
-        may (fun (i : index) -> iter_io (fun e -> f e.key e.value) i.io) t.index;
```

Comment:
why did you remove this line?

Reply:
the iter on index is called again a bit lower (https://github.com/mirage/index/blob/master/src/index.ml#L761), but git diff is not showing it.

Reply:
yes but why is this necessary to remove that line? Is that related to the change between replace and add?

Reply:
I think these tests are related to this PR, but I can move both the tests and the change to `iter` in a different PR, if you think is clearer.

Reply:
well that's a bit suspicious if we have to change the semantics of a function to make the tests pass :-) @pascutto do you remember why `iter` is iterating over the elements twice? That sounds like a bug?

Reply:
I'll check what's going on here.
A side note though: `iter` may hit some keys twice even if there are in fact no duplicates so it's not super robust to write tests assuming that `iter` reflects the correct state of duplication in the structure. (see our docstring here: https://github.com/mirage/index/blob/master/src/index.mli#L176-L183)